core/quantity/src/Formatter/Format.ts

```diff
@@ -176,6 +194,18 @@ export class BaseFormat {
         throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'stationSeparator' attribute. It should be an empty or one character string.`);
       this._stationSeparator = formatProps.stationSeparator;
     }
+
+    if (undefined !== formatProps.azimuthBase) { // optional; default is 0.0
```

Comment:
if the default is 0.0, shouldn't that be reflected in line 41 above, where the azimuthBasee property is defined? Same with azimuthCounterClockwise

Reply:
My thinking was, if I leave it "undefined" above and use a default in code, I can then tell if the value was explicitly set or not. So the serializer would know if it should serialize the value or not.
As alternatives, there could be a separate "isXXXExplicitlySet"-flag which we do often but I don't like that very much. Or I could not serialize if the value matches the default, but in that case the serialized value JSON could be different from the loaded value JSON and code might think it has changed even though it represents the same format....
So out of those options I chose this one..

Reply:
following up on this, now that we have serialization - is there an intention behind determining if it should serialize the value or not? Based on internal conversations we've also established what the default values should be right? The parser and formatter won't use these attributes anyways unless they are of type `Azimuth`?

Reply:
Coming from XML serialization, we do not guarantee, but try to preserve the contents of the file as much as we can. So if the spacer for example is set to " " in the file, even though that is the default, we try to remember that and serialize it back.
So far this has been a good practice, so if the input JSON does not contain an azimuthBase, and we serialize it back to Json, there should not be one. The default is 90, and if the input matches 90 we still want to serialize that back out again.
I like this approach. The code only uses the fields when the type is azimuth, but I'd still try and preserve them if they are not used.
Or would you like to change that? I do not have a strong opinion on this, I can be persuaded :-)

Reply:
I like that approach, it's just we already have existing fields that are created by default (like uom, deicmal separator, etc), so if the input json doesn't contain those fields, the output json would. If we've established default values for these 2 new properties, I think concretely putting it in the code will reflect that's the default we've established, rather than leaving the onus to the user to interpret and enter manually. I also don't have a strong opinion on this either so I'd rather have another person chime in 😄

Reply:
I changed some of the fields to have a getXXOrDefault getter alongside the normal getter, which will fill in the default if the value is undefined. For azimuthBase I couldn't do it because the default depends on the unit. But I think that's a good compromise.